### PR TITLE
refactor(docker-compose): .env variable substitution

### PR DESCRIPTION
**Pull request by caspiano**

### CODE REVIEW

**Comment by caspiano** [comment]
I've never used anchors, nice suggestion. The intent of this was to have the .env to centralise the config, this would allow us to substitute different .env for beta/pinned releases

**Comment by KimBurgess** [comment]
Oh yeah, definitely, I was not suggesting the usage of a `.env` file is dropped. This definitely makes sense to specify image tags in there for simple switching and can be combined with the usage of anchors.

The query was more specifically do we need items like `ES_HOST` and `ES_PORT` to ever come from an external config? These are likely to be the same across every variation of the environment.

**Comment by caspiano** [comment]
I believe we may use SaaS elastic instances at some point. The anchors sort of solve the problem I was trying to solve with those variables.

**Comment by caspiano** [comment]
I'll leave the more fine-grained env stuff in for the time being

File: docker-compose.yml [hunk 1]
```diff
@@ -57,13 +72,9 @@ services:
     depends_on:
```
Comment (KimBurgess) [inline docker-compose.yml:73]: Could potentially extend the anchors to cover dependencies too.

```yaml
x-rethink-client: &rethink-client
  depends_on:
    - rethink
  environment:
    - RETHINKDB_HOST: ${RETHINKDB_HOST:-rethink}
    - RETHINKDB_PORT: ${RETHINKDB_PORT:-28015}
    - RETHINKDB_DB:   ${RETHINKDB_DB:-place_development}
```

Which can then be merged into the top level of the service.

File: docker-compose.yml [hunk 2]
```diff
@@ -363,4 +341,4 @@ services:
...
       - elastic
     environment:
-      - HQ_DEFAULT_URL=http://${ELASTIC_HOST}:${ELASTIC_PORT}
+      HQ_DEFAULT_URL: "http://${ELASTIC_HOST}:${ELASTIC_PORT}"
```
Comment (KimBurgess) [inline docker-compose.yml:353]: It'd pedantic, but suggest that we stick to either quoted or non-quoted strings for consistent style. Currently this is mixed.

File: docker-compose.yml (hunk 1 above)
Comment (caspiano) [reply docker-compose.yml:73]: nice idea!

Comment (KimBurgess) [reply docker-compose.yml:73]: Might just be worth checking that it does a deep merge. The validity of that is pure speculation from my part.

**Pull request merged**